modules/overrides/config_test.go

```diff
@@ -306,3 +311,199 @@ func TestOverrides_AssertUserConfigurableOverridesAreASubsetOfRuntimeOverrides(t
 	err = d.Decode(&runtimeOverrides)
 	assert.NoError(t, err)
 }
+
+func TestFormatConversion(t *testing.T) {
+	legacyOverrides := generateTestLegacyOverrides()
+
+	// Verify that all fields have been populated in our test fixture
+	ensureAllFieldsPopulated(t, legacyOverrides)
+
+	// Convert to new format and back
+	newOverrides := legacyOverrides.toNewLimits()
+	convertedLegacyOverrides := newOverrides.toLegacy()
+
+	// Compare original and converted
+	assert.Equal(t, legacyOverrides, convertedLegacyOverrides)
+}
+
+// ensureAllFieldsPopulated checks that all fields in the struct have non-zero values
+// This helps catch if a new field is added to LegacyOverrides but not included in our test fixture
+func ensureAllFieldsPopulated(t *testing.T, o LegacyOverrides) {
+	v := reflect.ValueOf(o)
+	t.Helper()
+
+	// Get the type of the struct
+	structType := v.Type()
+
+	// Iterate through all fields
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i)
+		fieldName := structType.Field(i).Name
+
+		// Skip certain fields that can be zero in valid configs
+		switch fieldName {
+		case "IngestionArtificialDelay": // Can legitimately be nil
```

Comment:
what do you think about creating a static list of fieldNames, that way it would be easier to add new nil values to skip

```go
	skip := []string{"IngestionArtificialDelay"}

	if slices.Contains(skip, fieldName) {
		continue
	}
```

Reply:
For now it's just that field, but sounds good 👍